docs/docs/understanding/workflows/extending.md

```diff
@@ -0,0 +1,161 @@
+# Extending workflows
```

Comment:
Not sure if we want to show it, but another, quite fun way of customizing workflows is attaching steps to an existing one without subclassing. For example:
```
from somewhere import RAGWorkflow

# ad a simple function as a step to an existing workflow
@step(workflow=RAGWorkflow)
async def send_slack_notification(ev: StartEvent) -> None:
    async with httpx.AsyncClient() as client:
        await client.post('https://www.some_webook.com/', data={'msg': 'Workflow has started!'})


wf = RAGWorkflow()  # this will have the extra step `send_slack_notification` attached
await wf.run()
```

Reply:
This is a super cool example. We can probably add this in a the third upcoming docs pass on workflows 💪🏻